gap/examples.gi

```diff
@@ -370,3 +370,488 @@ GeneralisedPetersenGraphCons);
+          DigraphAddEdge(D, j, i);
+        fi;
+      fi;
+    od;
+  od;
+  return D;
+end);
+
+InstallGlobalFunction("DIGRAPHS_WindmillGraph",
+function(n, m)
+  local D, i, K, nrVert;
+  if (not IsInt(m)) or (not IsInt(n)) or m < 2 or n < 2 then
+    ErrorNoReturn("the arguments <n> and <m> must be integers greater than 1,");
+    return fail;
+  fi;
+  D := Digraph(IsMutableDigraph, []);
+  K := CompleteDigraph(n - 1);
+  nrVert := 1 + DigraphNrVertices(K) * m;
+  DigraphAddVertices(D, nrVert);
+  for i in [0 .. (m - 1)] do
+    DigraphAddEdges(D, DigraphEdges(K) + (i * DigraphNrVertices(K)));
+  od;
+  for i in [1 .. (DigraphNrVertices(D) - 1)] do
+    DigraphAddEdge(D, i, nrVert);
+    DigraphAddEdge(D, nrVert, i);
+  od;
+  return D;
+end);
+
+InstallGlobalFunction("DIGRAPHS_PathGraph",
+function(n)
+  local D;
+  if (not IsInt(n)) or n < 1 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 0,");
+    return fail;
+  fi;
+  D := Digraph(IsMutableDigraph, []);
+  DigraphAddVertices(D, n);
+  DigraphAddEdges(D, List([1 .. (n - 1)], x -> [x, x + 1]));
+  return DigraphSymmetricClosure(D);
+end);
+
+InstallGlobalFunction("DIGRAPHS_StackedPrismGraph",
+function(n, k)
+  if (not IsInt(n)) or (not IsInt(k)) or n < 3 or k < 1 then
+    ErrorNoReturn("the arguments <n> and <k> must be integers,
+      with <n> greater than 2 and <k> greater than 0,");
+    return fail;
+  fi;
+  return DigraphCartesianProduct(DIGRAPHS_CycleGraph(n),
+          DIGRAPHS_PathGraph(k));
+end);
+
+InstallGlobalFunction("DIGRAPHS_WebGraph",
+function(n)
+  local D, i;
+  if (not IsInt(n)) or n < 3 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 2,");
+    return fail;
+  fi;
+  D := DIGRAPHS_StackedPrismGraph(n, 3);
+  for i in [1 .. (n - 1)] do
+    DigraphRemoveEdge(D, i, i + 1);
+    DigraphRemoveEdge(D, i + 1, i);
+  od;
+  return D;
+end);
+
+InstallGlobalFunction("DIGRAPHS_HanoiGraph",
+function(n)
+  local D, C, i, nrVert, exp;
+  if (not IsInt(n)) or n < 1 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 0,");
+    return fail;
+  fi;
+  C := DIGRAPHS_CycleGraph(3);
+  D := C;
+  for i in [2 .. n] do
+    exp := 2 ^ (i - 2);
+    D := Digraph(IsMutableDigraph, []);
+    nrVert := DigraphNrVertices(C);
+    DigraphAddVertices(D, 3 * nrVert);
+    DigraphAddEdges(D, DigraphEdges(C));
+    DigraphAddEdges(D, DigraphEdges(C) + nrVert);
+    DigraphAddEdges(D, DigraphEdges(C) + (2 * nrVert));
+    DigraphAddEdge(D, nrVert / 2 + (1 / 2), nrVert + 1);
+    DigraphAddEdge(D, nrVert, 2 * nrVert + 1);
+    DigraphAddEdge(D, 2 * nrVert, DigraphNrVertices(D) - exp);
+    DigraphSymmetricClosure(D);
+
+    C := D;
+  od;
+  return D;
+end);
+
+InstallGlobalFunction("DIGRAPHS_BinomialTreeGraph",
+function(n)
+  local D, D_n, bits, i, is2n;
+  if (not IsInt(n)) or n < 1 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 0,");
+    return fail;
+  fi;
+  bits := Log(n, 2);
```

Comment:
I was looking at what's faster, here there are some timings:
```
gap> for n in [1..100000000] do t:=IsEvenInt(n) and IsPrimePowerInt(n);od;time;
15225
gap> for n in [1..100000000] do bits:=Log(n,2);is2n := n = 2 ^ bits;od;time;
19065
```

Reply:
Thank you, replaced existing check with faster one